packages/kit/src/components.ts

```diff
@@ -31,6 +31,11 @@ export async function addComponent (opts: AddComponentOptions) {
   await assertNuxtCompatibility({ nuxt: '>=2.13' }, nuxt)
   nuxt.options.components = nuxt.options.components || []
 
+  if (!opts.mode) {
+    const [, mode = 'all'] = opts.filePath.match(/\.(server|client)(\.\w+)*$/) || []
+    opts.mode = mode as 'all' | 'client' | 'server'
```

Comment:
As we use the mode quite a bit now ([plugins](https://github.com/nuxt/nuxt/blob/a073bec6b1b0bf82f435345306d7187f7f4f4b0f/packages/kit/src/plugin.ts#L37-L38) + components), I wonder if an extra type would help in case more modes will be added later on.

Reply:
actually, I remembered why it's this way - it's so we have a very visual guide to ensure the regexp matches the type assertion... there is already an actual type for `mode` that this 'checks' against